<launch>
    <!--Left Lidar 左激光雷达-->
    <node name="left_lidar" pkg="sick_tim" type="sick_tim551_2050001" respawn="false" output="screen">
        <param name="hostname" type="string" value="172.16.127.11"/>
        <param name="intensity" type="bool" value="false" />
        <param name="port" type="string" value="2112" />
        <param name="timelimit" type="int" value="5" />
        <param name="min_ang" type="double" value="-1.10" />
        <param name="max_ang" type="double" value="2.00" />
        <param name="frame_id" type="str" value="left_lidar"/>
        <remap from="scan" to="/left_lidar"/>
    </node>
    <node name="baselink_to_left_lidar" pkg="tf2_ros" type="static_transform_publisher" args="-0.11 0.28 -0.025 0 0.6111 -1.5714 frontwheel_right left_lidar" output="screen"/>
    
    <!--Right Lidar 右激光雷达-->
    <node name="right_lidar" pkg="sick_tim" type="sick_tim551_2050001" respawn="false" output="screen">
        <param name="hostname" type="string" value="172.16.127.10" />
        <param name="intensity" type="bool" value="false" />
        <param name="port" type="string" value="2112" />
        <param name="timelimit" type="int" value="5" />
        <param name="min_ang" type="double" value="-2.0796" />
        <param name="max_ang" type="double" value="1.1" />
        <param name="frame_id" type="str" value="right_lidar"/>
        <remap from="scan" to="/right_lidar"/>
    </node>
    <node name="baselink_to_right_lidar" pkg="tf2_ros" type="static_transform_publisher" args="-0.11 0.28 0.025 0 -0.6111 -1.5714 frontwheel_left right_lidar" output="screen"/>

    <!-- Sensor Fusion -->
    <node name="laser_scan_fusion" pkg="ros_whill" type="laser_scan_fusion">
        <param name="target_frame" value="base_link"/>
        <param name="fused_cloud_topic" value="/fused_cloud"/>
        <param name="scan0" value="/left_lidar"/>
        <param name="scan1" value="/right_lidar"/>
        <!--param name="scan2" value="/realsense_scan"/-->
    </node>

    <node name="fused_cloud_to_fused_scan" pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node">
        <remap from="cloud_in" to="/fused_cloud"/>
        <remap from="scan" to="/scan"/>
        <param name="angle_min" value="-3.14159265"/>
        <param name="angle_max" value="3.14159265"/>
        <param name="range_min" value="0"/>
        <param name="range_max" value="100"/>
    </node>
    
</launch>